### PR TITLE
tidb-in-kubernetes: add doc to restart tidb components on k8s

### PR DESCRIPTION
**Pull request by TomShawn**
### What is changed, added or deleted? 

Add a document to restart TiDB components on Kubernetes.

### What is the related PR or file link(s)? 

https://github.com/pingcap/docs-cn/pull/2156

### Which version does your change affect? 

dev, v3.0, v3.1

### CODE REVIEW

File: dev/tidb-in-kubernetes/maintain/restart.md [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+
+# Restart TiDB Cluster in Kubernetes
+
+This document describes how to forcibly restart the TiDB cluster in the Kubernetes cluster, including restarting a Pod, restarting all Pods of a component, and restarting all Pods of the TiDB cluster.
+
+> **Note:**
+>
+> TiDB Operator v1.0.x only supports restarting the Pod in a forcible way.
+>
+> - If the PD Pod being restarted is the PD Leader, its leadership does not automatically shift, causing the PD service to pause momentarily.
```
Comment (DanielZhangQD) [inline dev/tidb-in-kubernetes/maintain/restart.md]:
```suggestion
> - If the PD Pod being restarted is the PD Leader, its leadership does not automatically shift, causing the PD service to be unavailable momentarily.
```

File: dev/tidb-in-kubernetes/maintain/restart.md [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+> - If the TiKV Pod being restarted contains a Region Leader of the TiKV cluster, its leadership does not automatically shift, causing the request for accessing the corresponding data to fail.
```
Comment (DanielZhangQD) [inline dev/tidb-in-kubernetes/maintain/restart.md]:
```suggestion
> - If the TiKV Pod being restarted contains a Region Leader of the TiKV cluster, its leadership does not automatically shift, causing the requests of accessing the corresponding data to fail.
```

File: dev/tidb-in-kubernetes/maintain/restart.md [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+> - Restarting the TiDB Pod causes the request for accessing this Pod to fail.
```
Comment (DanielZhangQD) [inline dev/tidb-in-kubernetes/maintain/restart.md]:
```suggestion
> - Restarting the TiDB Pod causes the requests of accessing this Pod to fail.
```

File: TOC.md [hunk 4]
```diff
@@ -403,6 +403,7 @@
     - [Collect Logs](/tidb-in-kubernetes/maintain/log-collecting.md)
     - [Automatic Failover](/tidb-in-kubernetes/maintain/auto-failover.md)
     - [TiDB Binlog](/tidb-in-kubernetes/maintain/tidb-binlog.md)
+    - [Restart TiDB Cluster](/tidb-in-kubernetes/maintain/restart.md)
```
Comment (lilin90) [inline TOC.md]:
```suggestion
    - [Restart a TiDB Cluster](/tidb-in-kubernetes/maintain/restart.md)
```

File: tidb-in-kubernetes/maintain/restart.md [hunk 5]
```diff
@@ -0,0 +1,72 @@
+---
+title: Restart TiDB Cluster in Kubernetes
```
Comment (lilin90) [inline tidb-in-kubernetes/maintain/restart.md]: To keep it consistent with the `Monitor a TiDB Cluster in Kubernetes` doc.
```suggestion
title: Restart a TiDB Cluster in Kubernetes
```

File: tidb-in-kubernetes/maintain/restart.md [hunk 6]
```diff
@@ -0,0 +1,72 @@
...
+summary: Learn how to restart the TiDB cluster in the Kubernetes cluster.
```
Comment (lilin90) [inline tidb-in-kubernetes/maintain/restart.md]:
```suggestion
summary: Learn how to restart a TiDB cluster in the Kubernetes cluster.
```

File: tidb-in-kubernetes/maintain/restart.md [hunk 7]
```diff
@@ -0,0 +1,72 @@
...
+category: how-to
+---
+
+# Restart TiDB Cluster in Kubernetes
```
Comment (lilin90) [inline tidb-in-kubernetes/maintain/restart.md]:
```suggestion
# Restart a TiDB Cluster in Kubernetes
```

File: tidb-in-kubernetes/maintain/restart.md [hunk 8]
```diff
@@ -0,0 +1,72 @@
...
+
+This document describes how to forcibly restart the TiDB cluster in the Kubernetes cluster, including restarting a Pod, restarting all Pods of a component, and restarting all Pods of the TiDB cluster.
```
Comment (lilin90) [inline tidb-in-kubernetes/maintain/restart.md]:
```suggestion
This document describes how to forcibly restart a TiDB cluster in the Kubernetes cluster, including restarting a Pod, restarting all Pods of a component, and restarting all Pods of the TiDB cluster.
```

File: tidb-in-kubernetes/maintain/restart.md [hunk 9]
```diff
@@ -0,0 +1,72 @@
...
+
+> **Note:**
+>
+> TiDB Operator v1.0.x only supports restarting the Pod in a forcible way.
+>
+> - If the PD Pod being restarted is the PD leader, its leadership does not automatically shift, causing the PD service to be momentarily unavailable.
```
Comment (lilin90) [inline tidb-in-kubernetes/maintain/restart.md]: - Perhaps "temporarily" is more commonly used in our doc.
- In this note, we need to put emphasis on the result, so I suggest the following changes.

```suggestion
> - If the PD Pod being restarted is the PD leader, its leadership does not automatically shift, which can cause the PD service to be temporarily unavailable.
```

File: tidb-in-kubernetes/maintain/restart.md [hunk 10]
```diff
@@ -0,0 +1,72 @@
...
+> - If the TiKV Pod being restarted contains a Region leader of the TiKV cluster, its leadership does not automatically shift, causing the requests of accessing the corresponding data to fail.
```
Comment (lilin90) [inline tidb-in-kubernetes/maintain/restart.md]:
```suggestion
> - If the TiKV Pod being restarted contains a Region leader of the TiKV cluster, its leadership does not automatically shift, which can cause the requests of accessing the corresponding data to fail.
```

File: tidb-in-kubernetes/maintain/restart.md [hunk 11]
```diff
@@ -0,0 +1,72 @@
...
+> - Restarting the TiDB Pod causes the requests of accessing this Pod to fail.
```
Comment (lilin90) [inline tidb-in-kubernetes/maintain/restart.md]: Inappropriate tone, I think. To convey a kind of possibility.
```suggestion
> - Restarting the TiDB Pod can cause the requests of accessing this Pod to fail.
```

File: tidb-in-kubernetes/maintain/restart.md [hunk 12]
```diff
@@ -0,0 +1,72 @@
...
+
+{{< copyable "shell-regular" >}}
+
+```shell
+kubectl delete pod -n <namespace> -l app.kubernetes.io/component=<component-name>
+```
+
+> **Note:**
+>
+> - Replace `<component-name>` in the above commands with `pd`, `tidb`, or `tikv` to forcibly restart all Pods of the `PD`, `TiDB`, or `TiKV` component.
```
Comment (lilin90) [inline tidb-in-kubernetes/maintain/restart.md]:
```suggestion
> - To forcibly restart all Pods of the `PD`, `TiDB`, or `TiKV` component, replace `<component-name>` in the above commands with `pd`, `tidb`, or `tikv` respectively.
```

**Comment by TomShawn** [comment]
@lilin90 Comments addressed, PTAL again, thanks!

File: tidb-in-kubernetes/maintain/restart.md [hunk 13]
```diff
@@ -0,0 +1,72 @@
...
+# Restart a TiDB Cluster in Kubernetes
+
+This document describes how to forcibly restart a TiDB cluster in the Kubernetes cluster, including restarting a Pod, restarting all Pods of a component, and restarting all Pods of the TiDB cluster.
+
+> **Note:**
+>
+> TiDB Operator v1.0.x only supports restarting the Pod in a forcible way.
+>
+> - If the PD Pod being restarted is the PD leader, its leadership does not automatically shift, which can cause the PD service to be temporarily unavailable.
+> - If the TiKV Pod being restarted contains a Region leader of the TiKV cluster, its leadership does not automatically shift, which can cause the requests of accessing the corresponding data to fail.
```
Comment (DanielZhangQD) [inline tidb-in-kubernetes/maintain/restart.md]: From our current documentation, usually, we say leader `transfer` instead of `shift`.

**Comment by TomShawn** [comment]
@DanielZhangQD Comment addressed, PTAL again, thanks!

**Pull request merged**